cpu/atmega_common/periph/i2c.c

```diff
@@ -326,37 +265,37 @@ static int _start(uint8_t address, uint8_t rw_flag)
         DEBUG("ACK has been received for ADDRESS (read)\n");
     }
     else {
-        DEBUG("NOT ACK has been received for ADDRESS\n");
+        DEBUG("NACK has been received for ADDRESS\n");
         _stop();
         return -2;
     }
 
     return 0;
 }
 
+/* TODO : const uint8_t data instead of *data */
 static int _write(const uint8_t *data, int length)
 {
     for (int i = 0; i < length; i++) {
         /* Load DATA into TWDR Register.
          * Clear TWINT bit in TWCR to start transmission of data */
-        TWDR = data[i];
+        TWDR = data[0];
         TWCR = (1 << TWINT) | (1 << TWEN);
-        DEBUG("Byte %i transmitted\n", i+1);
+        DEBUG("Byte transmitted\n");
 
         /* Wait for TWINT Flag set. This indicates that DATA has been transmitted.*/
         while (!(TWCR & (1 << TWINT))) {}
 
         /* Check value of TWI Status Register. Mask prescaler bits. If status
          * different from MT_DATA_ACK, return number of transmitted bytes */
         if ((TWSR & 0xF8) != MT_DATA_ACK) {
-            DEBUG("NACK has been received for BYTE %i\n", i+1);
-            return i;
+            DEBUG("NACK has been received\n");
+            return 0;
         }
         else {
-            DEBUG("ACK has been received for BYTE %i\n", i+1);
-        }
+            DEBUG("ACK has been received\n");
+        } 
     }
-
     return length;
```

Comment:
I2C API functions does not return the length anymore. This should be replaced by 0